modules/aws/lb/README.md

```diff
@@ -0,0 +1,326 @@
+    <li><a href="#acknowledgments">Acknowledgments</a></li>
+  </ol>
+</details>
+
+
+<!-- USAGE EXAMPLES -->
+## Usage
+### Network Load Balancer with Target Group and Listener
+This example creates a Network Load Balancer (NLB) in AWS with associated target groups and listeners. The NLB is configured as internal-facing within the specified VPC private subnets. The target group is configured to use IP targets with TCP protocol on port 52110, with custom health check settings. A TCP listener is created on port 80 that forwards traffic to the target group. The module allows for configuration of cross-zone load balancing, deletion protection, and custom tags.
+```
+module "example_nlb" {
+  source = "github.com/zachreborn/terraform-modules//modules/aws/lb?ref=LATEST-VERSION-HERE"
+
+  # Load Balancer Configuration
+  name                             = "example-nlb"
+  internal                         = true
+  load_balancer_type               = "network"
+  subnets                          = module.vpc.private_subnet_ids
+  enable_cross_zone_load_balancing = false
+  enable_deletion_protection       = false
+
+  # Target Group Configuration
+  target_groups = {
+    main = {
+      name        = "example-tg1"
+      port        = 52110
+      protocol    = "TCP"
+      vpc_id      = module.vpc.vpc_id
+      target_type = "ip"
+      stickiness = [{
+        type = "source_ip"
+      }]
+      health_check = {
+        main = {
+          enabled             = true
+          healthy_threshold   = 3
+          interval            = 30
+          port                = "traffic-port"
+          protocol            = "TCP"
+          timeout             = 10
+          unhealthy_threshold = 3
+        }
+      }
+    }
+  }
+
+  # Listener Configuration
+  listeners = {
+    tcp = {
+      port     = 80
+      protocol = "TCP"
+      default_action = {
+        type = "forward"
+      }
+    }
+  }
+
+  tags = {
+    Environment = "some env"
+    Terraform   = "true"
+    Project     = "just a test"
+  }
+}
+```
+
+### Network Load Balancer with Target Group and Listener
+This example creates an Application Load Balancer (ALB) in AWS with associated target groups, listeners, and listener rules. The ALB is configured as internal-facing within the specified VPC private subnets and uses a security group for access control. The target group is configured to use instance targets with HTTP protocol on port 80, including health checks. An HTTP listener is created on port 80 that forwards traffic to the target group. The module also includes listener rules for source IP-based routing, allowing traffic from specific CIDR ranges (192.168.1.0/24 and 10.0.0.0/8). The configuration supports sticky sessions using load balancer cookies and allows for customization of health check parameters, security settings, and tags.
+```
+module "example_alb" {
+  source = "github.com/zachreborn/terraform-modules//modules/aws/lb?ref=dev_lb_maps"
```

Comment:
For the example, remove `?ref=dev_lb_maps` from the end.

Reply:
fixed